src/message/MessageContainer.js

```diff
@@ -64,7 +64,7 @@ export default class MessageContainer extends React.PureComponent {
       pushRoute,
     } = this.props;
     const MessageComponent = isBrief ? MessageBrief : MessageFull;
-    const childrenNodes = htmlToDomTree(message.match_content || message.content);
+    const childrenNodes = htmlToDomTree((message.match_content || message.content).replace(/\r\n/g, '<br />').replace(/[\r\n]/g, '<br />'));
```

Comment:
Yep, I think we're on the same page with this.  Can you open issues for the specific markup issues we want to fix?  I seem to recall there was one with the "inline URL preview" feature; were there others?